src/ops/tensor_ops.ts

```diff
@@ -93,6 +106,16 @@ function tensor<R extends Rank>(
   if (!isTypedArray(values) && !Array.isArray(values)) {
     values = [values] as number[];
   }
+
+  if(ENV.get('DEBUG')) {
```

Comment:
I'm not comfortable having this logic outside WebGL yet since it only affects WebGL for now. Can you move this to backend_webgl and check values in backend.write()